### ISSUE TITLE
influxDB-writer container keeps restarting continously.

### ISSUE THREAD

**Issue by howdyharry** [absmach/magistrala#759]
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).

**BUG REPORT**

1. What were you trying to achieve?
Trying to setup a mainflux instance and establish " Sending, receiving, storing to DB and visualizing data" by an user.
Following this tutorial:
[1](https://medium.com/mainflux-iot-platform/mainflux-open-source-iot-platform-set-up-and-usage-70bed698791a) and the manual [2](https://mainflux.readthedocs.io/en/latest/storage/#writers)

2. What are the expected results?
Writing incoming data to influxDB and visualizing with Grafana.

3. What are the received results?
Messages are sent and received by the things via cli but not written to influxdb and hence not visualized.
The influxdb-writer container just keeps restarting continously.
![Docker container status](https://user-images.githubusercontent.com/34143474/59339025-b6096c80-8d03-11e9-9eab-307d55ff8bb4.png)
![Mainflux status](https://user-images.githubusercontent.com/34143474/59339066-c883a600-8d03-11e9-9698-8feea25fcf68.png)
![Docker container logs for influxdb-writer](https://user-images.githubusercontent.com/34143474/59339093-d6d1c200-8d03-11e9-8034-b433448e4455.png)

4. What are the steps to reproduce the issue?
`git clone https://github.com/mainflux/mainflux.git`
`cd mainflux`
`docker-compose -f docker/docker-compose.yml -f docker/addons/influxdb-reader/docker-compose.yml -f docker/addons/influxdb-writer/docker-compose.yml up`
`docker  ps -a`

5. In what environment did you encounter the issue?
Ubuntu 16.04.6 LTS
Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false

Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:35:27 2019
 OS/Arch:           linux/amd64
 Experimental:      false

6. Additional information you deem important:
Via the CLI, the messages are sent and received by MQTT though. 
I reinstalled docker, docker-compose 3 thrice completely from scratch(from a bare system) over the last one week. The influxdb-writer docker-container still keeps restarting.

What makes the influxdb-writer container restart continously (apart from the restart flag in the yml file)?

**Comment by chombium** [comment]
@ howdyharry  Have you tried deleting the influxdb-writer container and running docker-compose again?

According to [this](https://stackoverflow.com/a/42260979) we might need to add ${PWD} to the [line](https://github.com/mainflux/mainflux/blob/master/docker/addons/influxdb-writer/docker-compose.yml#L57) where we mount the channels.toml file in the container.

Could you please try to change the line [57](https://github.com/mainflux/mainflux/blob/master/docker/addons/influxdb-writer/docker-compose.yml#L57) to be:
```
- ${PWD}/channels.toml:/config/channels.toml
```
and try again.

The thing with the ${PWD} is that depending on your terminal it may or may not be resolved

**Comment by howdyharry** [comment]
@chombium Yes, i deleted and ran docker-compose thrice already. It did not work. I now modified line 57 of the channels.toml to `- ${PWD}/channels.toml:/config/channels.toml`. Same issue. The influxdb-writer container still keeps restarting. But i can see a channels.toml folder created at the /mainflux/docker level.

**Comment by howdyharry** [comment]
I now ran docker-compose at mainflux/docker/addons/influxdb-writer level then it works.
`docker-compose -f ../../../docker/docker-compose.yml -f docker-compose.yml -f ../influxdb-reader/docker-compose.yml up`

**Comment by anovakovic01** [comment]
@howdyharry `channels.toml` file is more like example configuration. If you want to use it, you'll have to run influx composition from `docker/addons/influxdb-writer` directory.

**Comment by howdyharry** [comment]
@chombium @anovakovic01. Thanks it works.as suggested by @anovakovic01. Spent almost a week on this.  Excited to go forward with mainflux.
![Container Status](https://user-images.githubusercontent.com/34143474/59353257-dac10c80-8d22-11e9-8a13-2657a434a4e1.png)

**Comment by chombium** [comment]
@howdyharry I'm closing this as your problem has been fixed

**Comment by howdyharry** [comment]
@chombium - Moving forward the influxdb database stores only the latest message with the same timestamp always. No matter what. Even more, Grafana doesn`t show even the only latest message that is written to influxdb
![Postman message request](https://user-images.githubusercontent.com/34143474/59441819-71f89380-8df9-11e9-9942-c2d252ea4a11.png)

![influxdb-cli status](https://user-images.githubusercontent.com/34143474/59441855-7b81fb80-8df9-11e9-9e34-f1ff6b7cee28.png)

![Same timestamp with different values](https://user-images.githubusercontent.com/34143474/59442358-5fcb2500-8dfa-11e9-95e0-3a84a8b3802c.png)

![Retention policy](https://user-images.githubusercontent.com/34143474/59442377-69548d00-8dfa-11e9-9d93-0df8a4712624.png)

Am i missing a trick here?

**Comment by Khecari** [comment]
Hi,

I'm having the same problem, tried the same solution but to no avail. 
Only difference is i'm using ubuntu server 19.04 and running docker-composer with sudo

And then when i run docker-compose at mainflux/docker/addons/influxdb-writer i get
the following output:

![output](https://user-images.githubusercontent.com/48455462/63130190-a1598700-bf87-11e9-9dfc-72202c885a65.png)

I don't know how to proceed.

**Comment by dborovcanin** [comment]
@Khecari Can you, please, execute all the commands from the project root. For example `docker-compose -f docker/addons/influxdb-writer/docker-compose.yml up -d`?

**Comment by Khecari** [comment]
Hi, sorry for the delay, i just did a fresh install using Google Cloud with a Ubuntu Server 19.04

Commands:
//install docker
`$ sudo apt-get update`
`$ sudo apt-get install apt-transport-https ca-certificates curl gnupg-agent software-properties-common`
`$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add - ` 
`$ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"`
`$ sudo apt-get update`
`$ sudo apt-get install docker-ce docker-ce-cli containerd.io`

//install docker-compose
`$ sudo curl -L "https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose`
`$ sudo chmod +x /usr/local/bin/docker-compose`

//install mainflux with the issue 759 fix
`$ git clone https://github.com/mainflux/mainflux.git && cd mainflux`
//Double check that this weird sed thing actually changes the file ...
`$ sed -i '57s#.*#      - ${PWD}/channels.toml:/config/channels.toml#' docker/addons/influxdb-writer/docker-compose.yml`
//Executing commands from the proyect root
`$ sudo -E docker-compose -f docker/docker-compose.yml -f docker/addons/influxdb-reader/docker-compose.yml -f docker/addons/influxdb-writer/docker-compose.yml up -d`

The first time running docker-compose i got this

`ERROR: Network docker_mainflux-base-net declared as external, but could not be found. Please create the network manually using 'docker network create docker_mainflux-base-net' and try again.`

Then i did it again and the installation proceeded with no problems

Upon checking status with:
`
$ sudo docker ps --format "{{.Names}} {{.RunningFor}} {{.Status}}"`
I got:
mainflux-mqtt 11 minutes ago Up 11 minutes
mainflux-coap 11 minutes ago Up 11 minutes
mainflux-http 11 minutes ago Up 11 minutes
mainflux-ws 11 minutes ago Up 11 minutes
mainflux-things 11 minutes ago Up 11 minutes
mainflux-grafana 11 minutes ago Up 11 minutes
mainflux-users 11 minutes ago Up 11 minutes

> mainflux-influxdb-writer 11 minutes ago Restarting (1) 28 seconds ago

mainflux-normalizer 11 minutes ago Up 11 minutes
mainflux-influxdb 11 minutes ago Up 11 minutes
mainflux-es-redis 11 minutes ago Up 11 minutes
mainflux-ui 11 minutes ago Up 11 minutes
mainflux-things-redis 11 minutes ago Up 11 minutes
mainflux-users-db 11 minutes ago Up 11 minutes
mainflux-nginx 11 minutes ago Up 11 minutes
mainflux-influxdb-reader 11 minutes ago Up 11 minutes
mainflux-jaeger 11 minutes ago Up 11 minutes
mainflux-nats 11 minutes ago Up 11 minutes
mainflux-mqtt-redis 11 minutes ago Up 11 minutes
mainflux-things-db 11 minutes ago Up 11 minutes


And after seeing influxdb-writer logs with
`$ sudo docker logs mainflux-influxdb-writer`

I got:

2019/08/17 04:21:52 read /config/channels.toml: is a directory
2019/08/17 04:21:55 read /config/channels.toml: is a directory

**Comment by dborovcanin** [comment]
@Khecari I'm not sure why are you doing this step: `//Double check that this weird sed thing actually changes the file ...`. There are two reasons:

1. Unless you manually fix it, the indentation will be invalid, so your compose file will not be valid, as well.
2. Unless you have `channels.toml` in your project root, this won't work (because you're mapping unexisting file in docker-compose) - which, according to this error, seems to be the case.

Also, _please read the docs_: https://github.com/mainflux/mainflux/tree/master/writers/influxdb#using-docker-compose. If you want to modify things, please make sure that all the env vars and paths are set to valid values.

**Comment by Khecari** [comment]
@dusanb94 i'm not trying to do anything special, my problem is this issue's original problem.
I just clone the repo without modifying anything, and then proceed with the docker-compose in
the project root (i have no idea for what could i use the channels.toml ...)
And then i see that influxdb-writer keeps restarting and the message is about that channels.toml

I actually tried the same by using the 0.9.0 and 0.8.0 release and the problem persists
I tried copying the channels.toml file to the proyect root
I tried making a config directory with a channels.toml file inside
I tried all the above without modifyin the docker-compose.yml file.

I've been using mainflux for months, but i installed it in digitalocean with the docker image
in the market place. But now i switched to google cloud and had to install docker manually.

I know that's the cause but i have no clue on how to solve it.
